x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -503,20 +525,29 @@ export const LinksMenuUI = (props: LinksMenuProps) => {
 
     if (!isCategorizationAnomalyRecord) {
       // Add item from the start, but disable it during the URL generation.
-      const isLoading = openInDiscoverUrl === undefined;
+      const isLoading = discoverUrlError === undefined && openInDiscoverUrl === undefined;
 
       items.push(
         <EuiContextMenuItem
           key={`auto_raw_data_url`}
           icon="discoverApp"
-          disabled={isLoading}
+          disabled={discoverUrlError !== undefined || isLoading}
           href={openInDiscoverUrl}
           data-test-subj={`mlAnomaliesListRowAction_viewInDiscoverButton`}
         >
-          <FormattedMessage
-            id="xpack.ml.anomaliesTable.linksMenu.viewInDiscover"
-            defaultMessage="View in Discover"
-          />
+          {discoverUrlError ? (
```

Comment:
Can this action be hidden altogether if the Discover feature is not enabled in the current space? Here the disabled action and tooltip are displayed, but even if the data view is enabled the action will not work if the Discover feature is not enabled and this will be the result if you try and use the link:

![image](https://user-images.githubusercontent.com/7405507/152811713-c40c640b-011b-49aa-af19-e074174f1f88.png)

Reply:
Updated here https://github.com/elastic/kibana/pull/124408/commits/8969b796085b385cf7aad6d3d0e0297e2ed6eb11